Fix logging in parallel runspaces

Jaykul opened the pull request:
This fixes #6 and also fixes the module to work in PowerShell 6 and 7

JPRuskin commented on .gitignore:
```diff
@@ -214,3 +214,5 @@ pip-log.txt
 #Mr Developer
 .mr.developer.cfg
+
+Output/
```
Is there a reason we're using `Output/PowerShellLogging` rather than building to a versioned folder (given there was no build script previously, it seems this would assist with the workflow we've previously discussed).

Jaykul commented on Tests/Simple.Tests.ps1:
```diff
@@ -0,0 +1,63 @@
+#requires -Module @{ModuleName = "PowerShellLogging"; ModuleVersion = "1.3.0"}
```
You're right!

dlwyatt commented on HostIOInterceptor.cs:
```diff
@@ -127,9 +152,9 @@ public void DetachFromHost()
                                                             Collection<FieldDescription> descriptions)
         {
-            if (externalUI == null)
+            if (externalUI != null)
```
Looks like a little too much find/replace happened.  Several of these should still be `== null` :)

Jaykul replied:
Oh good grief. Good catch. The silly thing is, I pasted each one of those individually and customized the thrown messages 😝

Jaykul commented:
This last commit changes the build.ps1 script I added to accept a `Version` parameter (and by default use `gitversion` to calculate it) and build output to a folder with that as the name by default. That makes the dev versions of the module easier to test (if you make sure the parent to your project is in your `$Env:PSModulePath`). 

Having done that, I can also update all the version numbers in the build _output_ (in the library file and in the metadata), so I went ahead and did all of that in a single commit.

If you don't want that, I can revert that commit ...

Jaykul commented:
- The module now works in Invoke and in Runspaces without hosts now.
- It even works in remote runspaces without hosts (which is what I really needed _right now_).

However, his last commit demonstrates a remaining problem found in our QA:

Parallel runspaces in the same process still cause problems due to the 
`public static readonly HostIOInterceptor Instance`. 

Personally I'd like to see this merged, and will create a new issue for the other problem...

Jaykul commented:
While setting up an automated build against my branch I ran into some issues with the tests ...
Since the PR is still open, I've fixed those and added a couple showing the problem mentioned in #8

Jaykul commented:
FWIW, these changes passed our tests and QA and we're now working with a forked version ...

dlwyatt commented:
Cool. I’m on vacation for a couple of weeks. I can merge the PR now with my phone, but won’t be able to publish until I get back home
